<launch>
  <!-- ';' (semicolor) separated lists -->
  <arg name="ar_obj_list" default="chair.obj;ground_plane.obj" />
  <arg name="ar_mesh_scaling_list" default=".2;5.0" />
  <arg name="ar_controls_scaling_list" default="1.5;1.5" />
  <arg name="ar_mesh_initial_positions_list" default="0,0,-0.5;0,0,-0.0" />

  <!-- <arg name="ar_obj_list" default="cube.obj" />
  <arg name="ar_mesh_scaling_list" default="1.0" />
  <arg name="ar_controls_scaling_list" default="1.5" />
  <arg name="ar_mesh_initial_positions_list" default="0,0,-0.5" /> -->

<!--
  <arg name="ar_obj_list" default="simple_car.obj" />
  <arg name="ar_mesh_scaling_list" default="1.0" />
  <arg name="ar_controls_scaling_list" default="1.5" />
  <arg name="ar_mesh_initial_positions_list" default="0,0,-0.5" /> -->

  <!-- This is the camera calibration file (instrinsic calib). It is used for reprojection of 3d points on the image plane using camodocal's abstract camera -->
  <!-- <arg name="config_path" default="$(find cerebro)/config/mynteye_pinhole/mynteye_config.yaml" /> -->
  <!-- <arg name="config_path" default="$(find cerebro)/config/mynteye_kannala_brandt/mynteye_config.yaml" /> -->
  <arg name="config_path" default="$(find cerebro)/config/vinsfusion/realsense_d435i/left.yaml" />

 <node pkg="ar_demo" type="ar_demo_node3" name="ar_demo_node3_odom" output="screen">
    <!-- <remap from="~image_raw" to="/pg_17302081/image" doc="input image"/> -->
    <!-- <remap from="~camera_path" to="/pose_graph/pose_graph_camera_path"/> -->
    <!-- <remap from="~object_mesh_pose" to="/interactive_marker_server/object_mesh_pose"/> -->

    <param name="calib_file" type="string" value="$(arg config_path)"/>

    <param name="obj_list" type="string" value="$(arg ar_obj_list)"/>
    <param name="mesh_scaling_list" type="string" value="$(arg ar_mesh_scaling_list)"/>
    <param name="mesh_initial_positions_list" type="string" value="$(arg ar_mesh_initial_positions_list)" />

    <!-- <param name="raw_image_topic" type="string" value="/mynteye/left/image_raw"/> -->
    <param name="raw_image_topic" type="string" value="/camera/infra1/image_rect_raw"/>
    <param name="odometry_topic_name" type="string" value="/vins_estimator/camera_pose"/>
    <param name="ar_image_topic" type="string" value="AR_image_corrected"/>

</node>

<group if="0" >
    <node pkg="ar_demo" type="interactive_marker_server" name="interactive_marker_server" output="log" >
        <param name="obj_list" type="string" value="$(arg ar_obj_list)"/>
        <param name="mesh_scaling_list" type="string" value="$(arg ar_mesh_scaling_list)"/>
        <param name="controls_scaling_list" type="string" value="$(arg ar_controls_scaling_list)"/>
        <param name="mesh_initial_positions_list" type="string" value="$(arg ar_mesh_initial_positions_list)" />
    </node>
</group>

</launch>
